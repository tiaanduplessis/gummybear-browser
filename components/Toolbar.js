import React from 'react'
import { View, StyleSheet, TouchableOpacity } from 'react-native'
import { SimpleLineIcons } from '@expo/vector-icons'

import { TOOLBAR_HEIGHT } from '../constants/config'
import colors from '../constants/colors'

import { noop } from '../utils'

const Toolbar = ({ onHome, onBack, onForward, enableBack, enableForward }) => (
  <View style={styles.toolbar}>
    <TouchableOpacity
      onPress={onHome}
      hitSlop={{ top: 10, bottom: 10, left: 10, right: 10 }}
    >
      <SimpleLineIcons name='home' size={25} color={colors.black} />
    </TouchableOpacity>
    <View style={styles.innerContainer}>
      <TouchableOpacity onPress={onBack}>
        <SimpleLineIcons
          name='arrow-left'
          size={25}
          color={enableBack ? colors.black : colors.lightGray}
        />
      </TouchableOpacity>
      <TouchableOpacity onPress={onForward}>
        <SimpleLineIcons
          name='arrow-right'
          size={25}
          color={enableForward ? colors.black : colors.lightGray}
        />
      </TouchableOpacity>
    </View>
  </View>
)

Toolbar.defaultProps = {
  enableBack: false,
  enableForward: false,
  onHome: noop,
  onBack: noop,
  onForward: noop
}

const styles = StyleSheet.create({
  toolbar: {
    alignItems: 'center',
    padding: 12,
    zIndex: 1,
    justifyContent: 'space-between',
    flexDirection: 'row',
    shadowOffset: { width: 0, height: -1 },
    shadowColor: colors.gray,
    shadowOpacity: 0.7,
    elevation: 2,
    backgroundColor: colors.white,
    height: TOOLBAR_HEIGHT,
    borderColor: colors.offWhite,
    borderTopWidth: 1
  },
  innerContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    width: 100,
    justifyContent: 'space-between',
    alignItems: 'center'
  }
})

export default Toolbar
