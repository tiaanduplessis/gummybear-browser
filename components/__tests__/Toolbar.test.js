import 'react-native'
import React from 'react'
import Toolbar from '../Toolbar'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
    const tree = renderer.create(<Toolbar />).toJSON()
    expect(tree).toMatchSnapshot()
})
