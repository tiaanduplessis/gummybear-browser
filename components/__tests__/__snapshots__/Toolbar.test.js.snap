// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderColor": "#f6f6f6",
      "borderTopWidth": 1,
      "bottom": 0,
      "elevation": 2,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-between",
      "left": 0,
      "padding": 12,
      "position": "absolute",
      "right": 0,
      "shadowColor": "#757575",
      "shadowOffset": Object {
        "height": -1,
        "width": 0,
      },
      "shadowOpacity": 0.7,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 100,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
      />
    </View>
  </View>
</View>
`;
